lua/nvim-tree/explorer/sorters.lua

```diff
@@ -23,6 +23,31 @@ local function tbl_slice(t, first, last)
   return slice
 end
 
+---Evaluate `sort.folders_first` and `sort.files_first`
+---@param a table
+---@param b table
+---@param third_option_callback function|nil function should return `boolean`. Evaluated only if necessary
+---@return boolean|nil
+local function folders_or_files_first(a, b, third_option_callback)
```

Comment:
We could just use `comparator` for the third argument.

Reply:
Do you mean for the name of the variable instead of `third_option_callback`? If so I don't know if it's appropriate in this case since we aren't comparing anything, the function just returns the fallback value...

Reply:
`fallback` it is :)